node/narwhal/src/primary.rs

```diff
@@ -286,7 +298,7 @@ impl<N: Network> Primary<N> {
         // Broadcast the certified batch to all validators.
         self.gateway.broadcast(Event::BatchCertified(event));
 
-        info!("\n\n\n\n\nOur batch for round {} has been certified!\n\n\n\n", self.committee.round());
+        info!("\n\n\n\n\nOur batch {batch_id} for round {} has been certified!\n\n\n\n", self.committee.round());
```

Comment:
What's the rationale for printing the `batch_id`? (It is very verbose and uninformative)

Reply:
I don't agree on uninformative, this helps to correlate the different log lines. Otherwise it's not clear what has been certified (although it should maybe rather be debug level).

Reply:
This was instrumental in finding the race condition for the WARN. But now that you have a better fix for the quorum threshold, I think most of this PR can just be abandoned. Do you agree @vvp ?

Reply:
Yeah, I tested the latest fixes and they work better than this PR so let's just abandon this one.